CMakeLists.txt

```diff
@@ -113,6 +114,30 @@ if(NOT CUKE_DISABLE_GTEST)
 endif()
 
 #
+# Qt
+#
+
+if(NOT CUKE_DISABLE_QT)
+    find_package(Qt5Core)
+    find_package(Qt5Widgets)
+    find_package(Qt5Test)
+
+    if(${Qt5Core_FOUND} AND ${Qt5Widgets_FOUND} AND ${Qt5Test_FOUND})
+        message(STATUS "Found Qt version: ${Qt5Core_VERSION_STRING}")
+        set(QT_LIBRARIES Qt5::Core Qt5::Widgets Qt5::Test)
+    else()
+        find_package(Qt4 COMPONENTS QtCore QtGui QtTest)
+        if(QT4_FOUND)
+            set(QT_LIBRARIES Qt4::QtCore Qt4::QtGui Qt4::QtTest)
+            include(${QT_USE_FILE})
+        endif()
+    endif()
+    if(QT_LIBRARIES)
+        set(CMAKE_AUTOMOC ON)
```

Comment:
Unfortunately this causes it to add a ton of extra files to compile (one for every target). Can you remove this and instead add this to `examples/CalcQt/CMakeLists.txt`?
```cmake
set_target_properties(libcalcqt PROPERTIES AUTOMOC ON)
```